### DISCUSSION TITLE
Heterogeneous Graph Learning:GTN、GAT

### DISCUSSION THREAD

**Discussion by AlexKarry555** [pyg-team/pytorch_geometric#6899] [Q&A]
hello!, Dear professors. 
  In the process of learning, I made some changes to the GTN model. Some questions are raised about the experimental results.
  The original model is to learn the adjacency matrix, automatically generate meta-paths, and combine to generate a new adjacency matrix, then learn through a layer of GCN, and finally output through two linear layers.
  My change: replace GCN to GAT, but the experimental results were not ideal, from the original F1-score: 0.925 to 0.78The experimental effect of heads=1 is higher than heads=8, and GAT-layer=1 is better than GAT-layer=2.(The GAT model works similar whether it is defined or introduced from pyG)
  The loss function is also difficult to reduce, down to 0.42 after 100 epoch
  The dataset is used with the following data sets:ACM
Is GAT ineffective for meta-path-based learning?
How do you explain this reason?

**Comment by AlexKarry555** [comment]
![GTN](https://user-images.githubusercontent.com/63856164/224547011-8f9394e9-fb58-4c1b-99bd-3e31888f3889.jpg)

**Comment by rusty1s** [comment]
I sadly don't have a good answer for you, but I think it is important to note that different GNN layers have different strengths and biases built into them, which might explain your observed issues. For example, GCN is good at smoothing the surrounding neighborhood into a single embedding, while GAT can learn more fine-granular patterns (e.g., to discard certain neighbors).

**Reply by AlexKarry555** [reply]
Thank you very much for your reply!
According to your answer, I think that there are certain similarities and connections between the nodes on the metapath and the neighboring nodes. As you said, GCN is good at learning the embedding of surrounding neighbors, while GAT may lose the characteristics of some neighbor nodes, which may contradict the learning effect of the metapath.

**Reply by rusty1s** [reply]
Yeah, this is just a guess though, but usually metapaths are pretty dense, and so GCN may provide a better job of integrating the average signal of neighbors here.

**Reply by AlexKarry555** [reply]
Thank you very much for your guidance and help. I will continue to improve and learn from other aspects
